### PR TITLE
[TT-5321] Add documentation about new CRDs

### PR DESCRIPTION
**Pull request by buraksekili**
## Description
This PR updates the documentation, such that it will include details of the new CRDs for the Federation PoC.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] Make sure you are updating [CHANGELOG.md](../CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `gofmt -s -w .`
  - [ ] `go vet ./...`
  - [ ] `golangci-lint run`

### CODE REVIEW

File: config/samples/federation/README.md [hunk 1]
```diff
@@ -30,9 +33,132 @@ kubectl apply -f config/samples/federation/posts-subgraph.yaml
...
+
+An ApiDefinition must adhere to the following rules in order to represent an ApiDefinition for your SuperGraph CRDs.
+
+1. ApiDefinition and SuperGraph must be in the same namespace,
+2. `graphql.execution_mode` must be set to `supergraph`,
+3. `graphql.graph_ref` must be set to the name of the SuperGraph resource that you would like to refer.
+
+When you make an update on one of the subgraphs defined under your SuperGraph CRD's `subgraph_refs` field, Tyk Operator
+will update
+- Subgraph CRD,
```
Comment (komalsukhani) [inline config/samples/federation/README.md:151]: Should it be SuperGraph CR here?

File: config/samples/federation/README.md [hunk 2]
```diff
@@ -60,15 +186,15 @@ kubectl delete subgraphs.tyk.tyk.io <SUBGRAPH_NAME>
...
 subgraph in `graph_ref` field.
 
 ### SubGraph referenced in SuperGraph
 
-Although the subgraph is not referenced in any ApiDefinition, if it is referenced in the SuperGraph, Tyk Operator will 
+Although the subgraph is not referenced in any ApiDefinition, if it is referenced in the SuperGraph, Tyk Operator will
 not delete the subgraph again.
 
-In order to delete this subgraph, SuperGraph CRD should not have reference to corresponding subgraph in the `subgraphs_ref`.
+In order to delete this subgraph, SuperGraph CRD should not have reference to corresponding subgraph in the `subgraph_ref`.
```
Comment (komalsukhani) [inline config/samples/federation/README.md:197]:
```suggestion
In order to delete this subgraph, SuperGraph CR should not have reference to corresponding subgraph in the `subgraph_ref`.
```

File: config/samples/federation/README.md [hunk 3]
```diff
@@ -60,15 +186,15 @@ kubectl delete subgraphs.tyk.tyk.io <SUBGRAPH_NAME>
 
 If you have a subgraph which is referenced in any ApiDefinition, Tyk Operator will not delete the SubGraph.
 
-In order to delete this subgraph, corresponding ApiDefinition CRD must be updated, such that it has no reference to the 
+In order to delete this subgraph, corresponding ApiDefinition CRD must be updated, such that it has no reference to the
```
Comment (komalsukhani) [inline config/samples/federation/README.md:189]:
```suggestion
In order to delete this subgraph, corresponding ApiDefinition CR must be updated, such that it has no reference to the
```

File: config/samples/federation/README.md [hunk 4]
```diff
@@ -30,9 +33,132 @@ kubectl apply -f config/samples/federation/posts-subgraph.yaml
...
+- SuperGraph's ApiDefinition.
+
+Therefore, once you make an update on SubGraph CRD, you do not need to update your supergraph. It will be updated by Tyk Operator.
+
+With this approach, multiple teams can work on SubGraph CRDs and Tyk Operator will update the relevant SuperGraph ApiDefinition.
+
 ## Updating SubGraph
 
-Updating SubGraph is an easy operation with new CRDs. Once you update your SubGraph CRD, Tyk Operator will update your 
+Updating SubGraph is an easy operation with new CRDs. Once you update your SubGraph CRD, Tyk Operator will update your
```
Comment (komalsukhani) [inline config/samples/federation/README.md:161]:
```suggestion
Updating SubGraph is an easy operation with new CRDs. Once you update your SubGraph CR, Tyk Operator will update your
```

File: config/samples/federation/README.md (hunk 1 above)
Comment (buraksekili) [reply config/samples/federation/README.md:151]: yep, i'll add it

**Pull request merged**